test/e2e/app/app.go

```diff
@@ -527,9 +571,12 @@ func (app *Application) PrepareProposal(
 		// Coherence: No need to call parseTx, as the check is stateless and has been performed by CheckTx
 		txs = append(txs, tx)
 	}
+	var blob []byte
+	var exists bool
 	// Generate blob for the current height.
-	blob, exists := CreateBlob(req.Height)
-
+	if app.checkBlobHeight(req.Height, "prepare_proposal") {
+		blob, exists = CreateBlob(req.Height, app.cfg.BlobMaxBytesUpdateHeight)
```

Comment:
See [this comment](https://github.com/0xPolygon/cometbft/pull/19/files#r2095287605)